crates/starknet-devnet-server/src/api/json_rpc/mod.rs

```diff
@@ -508,6 +465,121 @@ pub enum JsonRpcRequest {
     #[serde(rename = "devnet_getConfig", with = "empty_params")]
     DevnetConfig,
 }
+
+impl JsonRpcRequest {
+    /// Should the request be retried by being forwarded to the forking origin?
+    fn is_forwardable_to_origin(&self) -> bool {
```

Comment:
this function is defined in error.rs file

Reply:
One is for errors, the other for requests. Not all requests require asking the forking origin in case we get an error, and of those that do require, not all errors require so. I did not introduce any new functionality, just refactored.